src/librustc_hir/Cargo.toml

```diff
@@ -16,6 +16,7 @@ rustc_data_structures = { path = "../librustc_data_structures" }
 rustc_index = { path = "../librustc_index" }
 rustc_span = { path = "../librustc_span" }
 rustc_serialize = { path = "../librustc_serialize" }
+rustc_session = { path = "../librustc_session" }
```

Comment:
Removing this dependency was a conscious decision, IIRC.

Reply:
Is there any documentation on which crates should/can depend on others?

There are only two functions in `librustc_hir` that call `check_name`, so avoiding this dependency hopefully won't be too hard.

Reply:
No documentations, https://github.com/rust-lang/rust/pull/68133 just broke dependencies that were easily breakable.

Reply:
I still think it would be good to break this dependency.

Reply:
I have removed the dependency by passing in closures that capture the `Session` to two functions.